### ISSUE TITLE
🚧Remove Assert Extensions

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/CASL#303]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Replace all `AssertExtensions` uses across the code base and then delete the `AssertExtensions` file.

**Reason:**
Over many months, all of the assertion code across the codebase has been getting refactored to use the **FluentAssertions** library API.  This has greatly improved the assertion code which has made the extension methods provided by the `AssertExtensions` class not required anymore.

> **Note**
> Most of the uses of the extensions methods will probably be replaced just by completing the rest
> of the issues for refactoring assertion code.  These issues are listed in the list of issue dependencies below.
> Please complete the issue dependencies below first before starting this issue.

### Acceptance Criteria

- [x] All uses of the `AssertExtensions` extensions methods replaced with **FluentAssertion** or standard XUnit code
- [x] The `AssertExtensions` file has been deleted from the code base.

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [x] Issue linked to the correct milestone _(if applicable)_.

### Issue Dependencies

- #105
- #106
- #107

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| CICD Changes        | `♻️cicd` |
| Config Changes      | `⚙️config` |
| Performance Improvements | `🏎️performance` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by thestbar** [comment]
Could you please assign this task to me?
